### ISSUE TITLE
Feature request: provide a way to fix "no matching version found inside the workspace" automatically

### ISSUE THREAD

**Issue by MichaelKapustey** [pnpm/pnpm#4477]
I deal with this pretty often. Resolving merge conflicts in package.json may lead to incorrectly typed version of dependency. Would be nice, if, after displaying message ' No matching version found for pkgname@1.2.12 inside the workspace' pnpm install would continue with 'found 1.3.5 instead. Would you like to update and proceed? Yes/no'

**Comment by vallerydelexy** [comment]
> 

not working 
![image](https://user-images.githubusercontent.com/17786532/212290293-9f9450ad-f72b-48f4-b291-bf694da94adf.png)

**Comment by Denny966** [comment]
#metoo

**Comment by maxlxq** [comment]
> > 
> 
> not working ![image](https://user-images.githubusercontent.com/17786532/212290293-9f9450ad-f72b-48f4-b291-bf694da94adf.png)

+1

**Comment by Azoraqua** [comment]
Me too.

**Comment by zkochan** [comment]
Why don't you just use the `workspace:*`, `workspace:~`, or `workspace:^` version spec to reference dependencies from the workspace? In that case the dependency will always be linked from the workspace and you'll never have to change this.

**Comment by Azoraqua** [comment]
> Why don't you just use the `workspace:*`, `workspace:~`, or `workspace:^` version spec to reference dependencies from the workspace? In that case the dependency will always be linked from the workspace and you'll never have to change this.

That would be the case if it worked, but for me (And apperantly a few others around here), it'll complain that the version mismatches or is not found at all. Which in turn prevents the use of pnpm almost altogether.

**Comment by craigmiller160** [comment]
I agree with the other comments. I'm not having this happen with a monorepo/workspace, I'm having this happen in general. I'm upgrading a library, and the newer version depends on a newer version of a previously installed library. the lockfile is preventing the upgrade.

**Comment by zkochan** [comment]
Until we have a reproduction available we can't fix it

**Comment by dcchuck** [comment]
Happy `pnpm` user here who ran into this issue.

@Azoraqua @craigmiller160 what are your version numbers? I'm using `YYYY.MM.DD-SNAPSHOT` for version numbering right now, as my modules are not quite ready for production.

This is what broke this process! It seems `pnpm` does some validation on the version number string - despite referencing it as `"workspace:*"` in my package.json.

I changed my version numbers to `0.1.0` and all was well. With guidance from PNPM on how they want this to work, I would be happy to patch.
